app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx

```diff
@@ -1,65 +1,110 @@
 import { format } from 'date-fns'
 import {
+  ALIGN_CENTER,
+  BORDERS,
+  Btn,
   COLORS,
   DIRECTION_COLUMN,
+  DIRECTION_ROW,
   Flex,
+  JUSTIFY_SPACE_BETWEEN,
   Icon,
-  SIZE_2,
+  NewPrimaryBtn,
   SPACING,
+  truncateString,
   TYPOGRAPHY,
 } from '@opentrons/components'
 import {
   useCreateRunMutation,
   useDeleteProtocolMutation,
   useProtocolQuery,
 } from '@opentrons/react-api-client'
-import { ProtocolMetadata } from '@opentrons/shared-data'
+import { ProtocolResource } from '@opentrons/shared-data'
+import { MAXIMUM_PINNED_PROTOCOLS } from '../../../App/constants'
 import {
-  BackButton,
-  QuaternaryButton,
-  TertiaryButton,
-} from '../../../atoms/buttons'
+  MediumButton,
+  TabbedButton,
+} from '../../../atoms/buttons/OnDeviceDisplay'
+import { Chip } from '../../../atoms/Chip'
 import { StyledText } from '../../../atoms/text'
+import { getPinnedProtocolIds, updateConfigValue } from '../../../redux/config'
 import { Deck } from './Deck'
 import { Hardware } from './Hardware'
 import { Labware } from './Labware'
+import { TooManyPinsModal } from '../TooManyPinsModal'
 
+import type { Dispatch } from '../../../redux/types'
 import type { OnDeviceRouteParams } from '../../../App/types'
 
-type ProtocolType = 'json' | 'python'
-type CreationMethod = 'Protocol Designer' | 'Python'
-
-const getCreationMethod = (protocolType: ProtocolType): CreationMethod =>
-  protocolType === 'json' ? 'Protocol Designer' : 'Python'
-
 const ProtocolHeader = (props: {
   title: string
-  date: number | null
-  protocolType: ProtocolType
   handleRunProtocol: () => void
 }): JSX.Element => {
+  const history = useHistory()
   const { t } = useTranslation(['protocol_info, protocol_details', 'shared'])
-  const { title, date, protocolType, handleRunProtocol } = props
+  const { title, handleRunProtocol } = props
+
+  const [truncate, setTruncate] = React.useState<boolean>(true)
+  const toggleTruncate = (): void => setTruncate(value => !value)
+
+  let displayedTitle = title
+
+  if (title.length > 92 && truncate) {
+    displayedTitle = truncateString(title, 92, 69)
+  }
   return (
-    <Flex flexDirection={DIRECTION_COLUMN} margin={SPACING.spacing5}>
-      <Flex gridGap={SPACING.spacing5} marginBottom={SPACING.spacing3}>
-        <StyledText as="h1">{title}</StyledText>
-        <TertiaryButton onClick={handleRunProtocol}>
-          {t('protocol_details:run_protocol')}
-        </TertiaryButton>
+    <Flex
+      alignItems={ALIGN_CENTER}
+      marginBottom={SPACING.spacingXXL}
+      justifyContent={JUSTIFY_SPACE_BETWEEN}
+    >
+      <Flex
+        alignItems={ALIGN_CENTER}
+        gridGap={SPACING.spacing5}
+        marginBottom={SPACING.spacing3}
+      >
+        <Btn onClick={() => history.goBack()}>
+          <Icon
+            name="back"
+            width="1.1875rem"
+            color={COLORS.darkBlack_hundred}
+          />
+        </Btn>
+        <Flex flexDirection={DIRECTION_COLUMN} maxWidth="42.625rem">
```

Comment:
Seems that we need to add space(8px) between chip and text from design.

```suggestion
        <Flex flexDirection={DIRECTION_COLUMN} maxWidth="42.625rem" gridGap={SPACING.spacing3}>
```